app/controllers/products_controller.rb

```diff
@@ -40,15 +37,7 @@ def update
     end
   end
 
-  def destroy
-    show
-    @product.destroy
-
-    redirect_to products_path
-  end
-
   def retire
-    @product = Product.find(params[:id])
     @product.retire_toggle!
     @product.save
```

Comment:
I think that the `save` logic should be encapsulated in the `retire_toggle!` method - especially because it has been specified as a `!` method